File: exercise/week04/ts/src/construct/stealthCloakUpdater.ts
import { Artefact } from "./artefact";
import { ArtefactUpdater } from "./artefactUpdater";

export class StealthCloakUpdater implements ArtefactUpdater {
    update(artefact: Artefact): void {
        artefact.timeToLive--;
        if (artefact.timeToLive % 2 === 0) {
            artefact.integrity = Math.max(0, artefact.integrity - 1);
        }
    }
}

File: exercise/week04/java/src/main/java/construct/StealthCloakUpdater.java
package construct;

public class StealthCloakUpdater implements ArtefactUpdater {
    @Override
    public void update(Artefact artefact) {
        artefact.timeToLive--;
        if (artefact.timeToLive % 2 == 0) {
            artefact.integrity = Math.max(0, artefact.integrity - 1);
        }
    }
}